multiregion: TestMultiRegionDataDriven/regional_by_table fails

At 049aff16460183cbd8f0b9ca922e9690f89779d4:

```
# make stress PKG=./pkg/ccl/multiregionccl TESTS=TestMultiRegionDataDriven/regional_by_table
Running make with -j12
GOPATH set to /go
mkdir -p lib
rm -f lib/lib{geos,geos_c}.so
cp -L /go/native/x86_64-linux-gnu/geos/lib/lib{geos,geos_c}.so lib
GOFLAGS= go test   -mod=vendor -exec 'stress ' -tags ' gss make x86_64_linux_gnu crdb_test' -ldflags '-X github.com/cockroachdb/cockroach/pkg/build.typ=development -extldflags "" -X "github.com/cockroachdb/cockroach/pkg/build.tag=v21.2.0-alpha.00000000-4558-g049aff1646" -X "github.com/cockroachdb/cockroach/pkg/build.rev=049aff16460183cbd8f0b9ca922e9690f89779d4" -X "github.com/cockroachdb/cockroach/pkg/build.cgoTargetTriple=x86_64-linux-gnu"  ' -run "TestMultiRegionDataDriven/regional_by_table" -timeout 0 ./pkg/ccl/multiregionccl  -v -args -test.timeout 45m
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s

initialized metamorphic constant "coldata-batch-size" with value 96
initialized metamorphic constant "kv.rangefeed.catchup_scan_iterator_optimization.enabled" with value true
initialized metamorphic constant "kv-batch-size" with value 1
initialized metamorphic constant "row-container-rows-per-chunk-shift" with value 1
initialized metamorphic constant "zig-zag-joiner-batch-size" with value 1
initialized metamorphic constant "spilling-queue-initial-len" with value 7
initialized metamorphic constant "merge-joiner-groups-buffer" with value 5
initialized metamorphic constant "max-batch-size" with value 5628
initialized metamorphic constant "max-batch-byte-size" with value 22242425
initialized metamorphic constant "async-IE-result-channel-buffer-size" with value 24
initialized metamorphic constant "restore-worker-concurrency" with value 1
I210928 18:35:36.372027 1 (gostd) rand.go:104  [-] 1  random seed: 7260279534851305329
=== RUN   TestMultiRegionDataDriven
    test_log_scope.go:79: test logs captured to: /tmp/logTestMultiRegionDataDriven123061820
    test_log_scope.go:80: use -show-logs to present logs inline
=== RUN   TestMultiRegionDataDriven/regional_by_table
    datadriven_test.go:122: 
        testdata/regional_by_table:1:
        new-cluster [1 args]
        <no input to command>
        ----
    datadriven_test.go:122: 
        testdata/regional_by_table:4:
        exec-sql [1 args]
        CREATE DATABASE db PRIMARY REGION "us-east-1" REGIONS  "us-central-1", "eu-central-1";
        ----
    datadriven_test.go:122: 
        testdata/regional_by_table:8:
        exec-sql [1 args]
        CREATE TABLE db.rbt(k INT PRIMARY KEY, v INT) LOCALITY REGIONAL BY TABLE IN "us-east-1";
        ----
    datadriven_test.go:122: 
        testdata/regional_by_table:12:
        exec-sql [1 args]
        INSERT INTO db.rbt VALUES (1, 1), (2,2)
        ----
    datadriven_test.go:122: 
        testdata/regional_by_table:16:
        wait-for-zone-config-changes [8 args]
        <no input to command>
        ----
    datadriven_test.go:122: 
        testdata/regional_by_table:19:
        trace-sql [1 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        served locally: true
        served via follower read: false
    datadriven_test.go:122: 
        testdata/regional_by_table:25:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:122: 
        testdata/regional_by_table:30:
        trace-sql [1 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        served locally: false
    datadriven_test.go:122: 
        testdata/regional_by_table:35:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 2
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:122: 
        testdata/regional_by_table:40:
        trace-sql [1 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        served locally: false
    datadriven_test.go:122: 
        testdata/regional_by_table:46:
        sleep-for-follower-read [0 args]
        <no input to command>
        ----
    datadriven_test.go:122: 
        testdata/regional_by_table:49:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:122: 
        testdata/regional_by_table:55:
        trace-sql [1 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        served locally: false
    datadriven_test.go:122: 
        testdata/regional_by_table:60:
        trace-sql [1 args]
        SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        ----
        served locally: false
    datadriven_test.go:122: 
        testdata/regional_by_table:65:
        refresh-range-descriptor-cache [2 args]
        SELECT * FROM db.rbt WHERE k = 2
        ----
        LAG_BY_CLUSTER_SETTING
    datadriven_test.go:122: 
        testdata/regional_by_table:71:
        trace-sql [1 args]
        SELECT * FROM db.rbt WHERE k = 1
        ----
        served locally: false
    datadriven_test.go:122: 
        testdata/regional_by_table:76:
        trace-sql [1 args]
        SELECT * FROM db.rbt AS OF SYSTEM TIME follower_read_timestamp() WHERE k = 1
        ----
        served locally: true
        served via follower read: true
    datadriven_test.go:122: 
        testdata/regional_by_table:83:
        exec-sql [1 args]
        ALTER DATABASE db ADD REGION "us-west-1";
        ----
    datadriven_test.go:122: 
        testdata/regional_by_table:88:
        exec-sql [1 args]
        ALTER TABLE db.rbt SET LOCALITY REGIONAL BY TABLE IN "us-west-1";
        ----
    datadriven_test.go:122: 
        testdata/regional_by_table:92:
        wait-for-zone-config-changes [6 args]
        <no input to command>
        ----
    datadriven_test.go:122: 
        testdata/regional_by_table:96: SELECT * FROM db.rbt WHERE k = 1
        expected:
        served locally: true
        served via follower read: false
        
        found:
        recording contains > 1 dist sender send messages
=== CONT  TestMultiRegionDataDriven
    datadriven_test.go:390: -- test log scope end --
test logs left over in: /tmp/logTestMultiRegionDataDriven123061820
--- FAIL: TestMultiRegionDataDriven (14.72s)
    --- FAIL: TestMultiRegionDataDriven/regional_by_table (14.70s)
```

--- a/pkg/ccl/multiregionccl/testdata/regional_by_table
+++ b/pkg/ccl/multiregionccl/testdata/regional_by_table
@@ -92,6 +92,11 @@ ALTER TABLE db.rbt SET LOCALITY REGIONAL BY TABLE IN "us-west-1";
 wait-for-zone-config-changes db-name=db table-name=rbt num-voters=3 num-non-voters=3 leaseholder=3 voter=4,5
 ----
 
+refresh-range-descriptor-cache idx=3 table-name=rbt
+SELECT * FROM db.rbt WHERE k = 2
+----
+LAG_BY_CLUSTER_SETTING
+
 # Reads from us-west-1 now should be local since we're homed in us-west-1.
 trace-sql idx=3
 SELECT * FROM db.rbt WHERE k = 1

